### ISSUE TITLE
RUF001 reports false positive for non-latin languages

### ISSUE THREAD

**Issue by berzi** [charliermarsh/ruff#4190]
The RUF001 rule will detect individual unicode non-latin-alphabet characters as ambiguous even when they are part of a word or string which is entirely comprised of non-latin-alphabet characters. This results in lots of false positives in strings in other languages, and autofix (if enabled with "ALL", for example) will destructively change these strings.

The rule should be improved to be a bit more nuanced, or else it will have to be disabled on any codebase which contains other alphabets (I'm not adding noqa comments on each line where Ruff thinks г should be r).

Examples:
```py
# Afghanistan's name in Cyrillic
"Афганистан"  # RUF001 String contains ambiguous unicode character `г` (did you mean `r`?)
# Most other letters are also signalled, since they look similar to latin letters.

# And in Arabic:
"أفغانستان" # RUF001 String contains ambiguous unicode character `ا` (did you mean `l`?)

# Zimbabwe in Armenian:
"Զիմբաբվե"  # RUF001 String contains ambiguous unicode character `ա` (did you mean `w`?)

My current version of Ruff is 0.0.263.
